ambry-cloud/src/main/java/com.github.ambry.cloud/azure/CosmosChangeFeedBasedReplicationFeed.java

```diff
@@ -0,0 +1,256 @@
+      this.cacheSessionId = cacheSessionId;
+      this.fetchedEntries = fetchedEntries;
+      this.partitionId = partitionId;
+      this.creationTimestamp = System.currentTimeMillis();
+    }
+
+    /**
+     * Return start continuation token.
+     * @return start continuation token.
+     */
+    String getStartContinuationToken() {
+      return startContinuationToken;
+    }
+
+    /**
+     * Return the end continuation token.
+     * @return end continuation token.
+     */
+    String getEndContinuationToken() {
+      return endContinuationToken;
+    }
+
+    /**
+     * Return the Azure request id.
+     * @return Azure request id.
+     */
+    String getCacheSessionId() {
+      return cacheSessionId;
+    }
+
+    /**
+     * Return the fetch entries list.
+     * @return {@link List} of {@link CloudBlobMetadata} entries.
+     */
+    List<CloudBlobMetadata> getFetchedEntries() {
+      return fetchedEntries;
+    }
+
+    /**
+     * Return the {@code partitionId}
+     * @return {@code partitionId}
+     */
+    public String getPartitionId() {
+      return partitionId;
+    }
+
+    /**
+     * Return the creation time stamp.
+     * @return  {@code creationTimestamp}
+     */
+    public long getCreationimestamp() {
+      return creationTimestamp;
+    }
+
+    /**
+     * Check if is this entry is expired. The condition for expiry uses {@code creationTimestamp}. This is good enough as
+     * this means that the cached set of fetches entries hasn't been consumed within the invalidation duration.
+     * @return true if this entry is expired. false otherwise.
+     */
+    boolean isExpired() {
+      return creationTimestamp < System.currentTimeMillis() - CACHE_INVALIDATION_DURATION_IN_MILLIS;
+    }
+  }
+
+  // change feed cache by cache session id
+  private final ConcurrentHashMap<String, ChangeFeedCacheEntry> changeFeedCache;
+  private final int defaultCacheSize;
+  private final CosmosDataAccessor cosmosDataAccessor;
+  private final AzureMetrics azureMetrics;
+  private final static long CACHE_INVALIDATION_DURATION_IN_MILLIS = 60 * 60 * 1000; //1 hour
+
+  /**
+   * Constructor to create a {@link CosmosChangeFeedBasedReplicationFeed} object.
+   * @param cosmosDataAccessor {@link CosmosDataAccessor} object.
+   * @param azureMetrics{@link {@link AzureMetrics} object.
+   */
+  public CosmosChangeFeedBasedReplicationFeed(CosmosDataAccessor cosmosDataAccessor, AzureMetrics azureMetrics) {
+    this.defaultCacheSize = AzureCloudDestination.getFindSinceQueryLimit();
+    changeFeedCache = new ConcurrentHashMap<>();
+    this.cosmosDataAccessor = cosmosDataAccessor;
+    this.azureMetrics = azureMetrics;
+    // schedule periodic invalidation of cache
+    Utils.newScheduler(1, false)
+        .scheduleAtFixedRate(() -> changeFeedCache.entrySet().removeIf(entry -> entry.getValue().isExpired()),
+            CACHE_INVALIDATION_DURATION_IN_MILLIS, CACHE_INVALIDATION_DURATION_IN_MILLIS, TimeUnit.MILLISECONDS);
```

Comment:
I wonder why we choose `CACHE_INVALIDATION_DURATION_IN_MILLIS` as initial delay. Can we use a random initial delay?